osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -126,9 +126,9 @@ public override bool HandleScale(Vector2 scale, Anchor anchor)
             return true;
         }
 
-        public override bool HandleFlip(Direction direction)
+        public override bool HandleFlip(Direction direction, bool flipOverOrigin)
         {
-            var selectionQuad = getSelectionQuad();
+            var selectionQuad = flipOverOrigin ? ScreenSpaceDrawQuad : getSelectionQuad();
```

Comment:
This isn't exactly right either. `ScreenSpaceDrawQuad` is taken from the `SkinSelectionHandler` which is bigger than the active preview area:

![osu_2022-01-05_20-39-54](https://user-images.githubusercontent.com/20418176/148278692-8e4d4ec8-dff3-44ac-b88f-bfc9f3541d9f.jpg)

causing some rather strange results when a selection that spans the entire width of the screen is flipped:

https://user-images.githubusercontent.com/20418176/148278760-2d5cba69-bb17-4b8b-b4d8-a69585f8836a.mp4

Reply:
Figured that might be the case. I've changed to using the direct parent of each drawable for now. Maybe not 100% correct, but does work for all existing cases.

Reply:
Not sure how long that'll hold up but it does appear to do the job for now.